test/provider/upgrade/upgrade_test.go

```diff
@@ -148,11 +146,33 @@ func TestProviderUpgrade(t *testing.T) {
 			},
 		},
 	}
-	for name, tc := range cases {
-		t.Run(name, func(t *testing.T) {
-			if err := tc.body(); err != nil {
-				t.Fatal(err)
-			}
-		})
+
+	config := getConfiguration()
+
+	for _, pr := range config.Providers {
+		for name, tc := range cases {
+			t.Run(name, func(t *testing.T) {
+				if err := tc.body(pr); err != nil {
+					t.Fatal(err)
+				}
+			})
+		}
+	}
+
+}
+
+func getConfiguration() c.Configurations {
```

Comment:
Some of these functions could likely live with the `provider` conformance API types